commons/src/main/java/io/aiven/kafka/tieredstorage/commons/io/SegmentCompressionChecker.java

```diff
@@ -0,0 +1,28 @@
+package io.aiven.kafka.tieredstorage.commons.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Function;
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.FileRecords;
+
+/**
+ * Checks if segment requires compression or not.
+ * To be used when segment files are received on archival.
+ */
+public class SegmentCompressionChecker implements Function<File, Boolean> {
+
+    /**
+     * @param file Kafka log segment file
+     * @return true if requires compression, else false
+     */
+    @Override
+    public Boolean apply(File file) {
```

Comment:
```suggestion
    public boolean apply(final File file) {
```

Also, maybe some more concrete name, like `check`?

Reply:
it's like this because it is implementing `Function<File, Boolean>`. Should we fallback to plain class?

Reply:
Yeah, probably `Function` doesn't give us much